Alex Birillo

biralex-bsuir opened the pull request:
I am very stupid and apologize for that. I will try to correct all errors you indicated

biralllex committed:
add all files for iteration 1 and 2

biralllex committed:
add all files for iteration 1 and 2

AlexeiBuzuma commented on rss_reader/arg.py:
```diff
@@ -0,0 +1,9 @@
+import argparse
+
+parser = argparse.ArgumentParser(description = "simple rss-reader")
```
в глобальной секции не должно содержаться никакой логики. советую перенести парсинг аргументов в отдельную функцию

AlexeiBuzuma commented on rss_reader/news.py:
```diff
@@ -0,0 +1,93 @@
+import re
+import html
+import json
+from arg import *
```
Очень не рекомендуется использовать импорт через *

AlexeiBuzuma commented on rss_reader/news.py:
```diff
@@ -0,0 +1,93 @@
+        if not args.limit:
+        	prepared_news.append(newsdict)
+    with open("data.json", "w") as output:
+    	json.dump(prepared_news, output, indent = 3, ensure_ascii = False)
```
Разный размер табуляции

AlexeiBuzuma commented on rss_reader/rss_reader.py:
```diff
@@ -0,0 +1,26 @@
+	if args.version:
+		print("Current version: " + version.version)
+
+	if (args.verbose):
```
советую посмотреть в сторону модуля `logging`

biralllex committed:
[WIP] Alex Birillo
Almost all of comments have been fixed. Logs are added through logging.
Logic moved to functions.

biralllex committed:
[WIP] Alex Birillo
Merge branch 'generalTask' of https://github.com/biralex-bsuir/PythonHomework into generalTask

biralllex committed:
Functions for iteration 3 added
Implemented caching and search by date

biralllex committed:
fixed bug related to --help
add to README.md description related to --date

AlexeiBuzuma commented on setup.py:
```diff
@@ -0,0 +1,33 @@
+with open("README.md", "r") as f:
+    long_description = f.read()
+
+setuptools.setup(
```
при запуске в докер контейнере python:latest падает вот с такой ошибкой 
```bash
# rss_reader https://news.tut.by/rss/geonews/brest.rss
Traceback (most recent call last):
  File "/usr/local/bin/rss_reader", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/site-packages/rss_reader/rss_reader.py", line 24, in main
    news.caching_news(captured_news)
  File "/usr/local/lib/python3.8/site-packages/rss_reader/news.py", line 57, in caching_news
    owner_name = os.getlogin()
OSError: [Errno 6] No such device or address
```

biralllex committed:
Improved docker compatibility via different location of the cash files

biralllex committed:
Added files for iteration №4

biralllex committed:
Necessary for correct work modules are added to setup.py

biralllex committed:
Files added for the fifth iteration
Updated README.md to the current state

biralllex committed:
Fixed a bug with painting over unnecessary areas. README.md updated

biralllex committed:
Some fixes for GMT news

AlexeiBuzuma commented on rss_reader/news.py:
```diff
@@ -0,0 +1,212 @@
+            except EOFError:
+                break
+
+def compare_dates(newsdate):
```
Док стринги очень полезная вещь. Есть много функций, которые не содержат документации совсем :(

AlexeiBuzuma commented on rss_reader/news.py:
```diff
@@ -0,0 +1,212 @@
+        news_date = datetime.strptime(newsdate[5:len(newsdate)-13], "%d %b %Y")
+    news_date = str(news_date)[:10]
+    requared_date = arg.args.date
+    requared_date = requared_date[0:4] + "-" + requared_date[4:6] + "-" + requared_date[6:8]
```
очень много магических чисел. очень сложно понять, почему они тут есть и что они значат

AlexeiBuzuma commented on rss_reader/news.py:
```diff
@@ -0,0 +1,212 @@
+    return news_date == requared_date
+
+def find_news_in_cache():
+    isNewsFind = False
```
Не весь код соответствует pep8

AlexeiBuzuma commented on rss_reader/rss_fb2.py:
```diff
@@ -0,0 +1,89 @@
+def encode_image(path):
+    """Gets the path to the image and returns it into encoded string"""
+    with open(path, "rb") as im:
+                    encoded_image = base64.b64encode(im.read())
```
Немного съехала табуляция

AlexeiBuzuma commented on rss_reader/news.py:
```diff
@@ -0,0 +1,212 @@
+    else:
+        file_path = """C:\\news.rss"""
+
+    with open(file_path, "rb") as input:
```
имя `input` существует изначально в build-in (функция, которая позволяет считать ввод с клавиатуры) 
лучше не переопределяь уже существующие имена, это может иногда сильно запутывать

AlexeiBuzuma commented on test/test.py:
```diff
@@ -0,0 +1,27 @@
+from rss_reader.rss_fb2 import encode_image
+from rss_reader import arg
+
+class handlers(unittest.TestCase):
```
Слабое покрытие юнит-тестами

AlexeiBuzuma commented on rss_reader/rss_pdf.py:
```diff
@@ -0,0 +1,95 @@
+        img_file.write(img.content)
+        img_file.close()
+
+def remove_image(i):
```
что такое? image_name или image_index? 
лучше не создавать имена в один символ, такой код тяжело читается

AlexeiBuzuma commented on LICENSE:
```diff
@@ -0,0 +1,20 @@
+
```
Коммит сообщения должны нести информацию о том, что сделано внутри этого коммита. Например сообщение "[WIP] Alex Birillo" в целом не несет в себе никакой информацию.

AlexeiBuzuma commented on rss_reader/news.py:
```diff
@@ -0,0 +1,212 @@
+from colored import fore, style
+
+
+class News():
```
Скобки можно опустить

AlexeiBuzuma commented on rss_reader/news.py:
```diff
@@ -0,0 +1,212 @@
+def write_to_json(news):
+    """Creates data.json document in the directory which the application was launched"""
+    prepared_news = []
+    for i in range(len(news)):
```
`for index, item in enumerate(news):`
